client/client.go

```diff
@@ -127,12 +141,102 @@ func (client Mikrotik) UpdateDnsRecord(id, name, address string, ttl int) error
 
 func (client Mikrotik) DeleteDnsRecord(id string) error {
 	c, err := client.getMikrotikClient()
+
+	if err != nil {
+		return err
+	}
 	cmd := strings.Split(fmt.Sprintf("/ip/dns/static/remove =numbers=%s", id), " ")
 	fmt.Println(fmt.Sprintf("[INFO] Running the mikrotik command: `%s`", cmd))
 	_, err = c.RunArgs(cmd)
 	return err
 }
 
+func (client Mikrotik) AddDhcpLease(address, macaddress string) (*routeros.Reply, error) {
+	c, err := client.getMikrotikClient()
+
+	if err != nil {
+		return nil, err
+	}
+	cmd := strings.Split(fmt.Sprintf("/ip/dhcp-server/lease/add =address=%s =mac-address=%s", address, macaddress), " ")
+	fmt.Println(fmt.Sprintf("[INFO] Running the mikrotik command: `%s`", cmd))
+	r, err := c.RunArgs(cmd)
+	return r, err
+}
+
+func (client Mikrotik) FindDhcpLease(address string) (*DhcpLease, error) {
+	c, err := client.getMikrotikClient()
+
+	if err != nil {
+		return nil, err
+	}
+	cmd := "/ip/dhcp-server/lease/print"
+	fmt.Println(fmt.Sprintf("[INFO] Running the mikrotik command: `%s`", cmd))
+	r, err := c.Run(cmd)
+	found := false
+	var sentence *proto.Sentence
+
+	if err != nil {
+		return nil, err
+	}
+
+	for _, reply := range r.Re {
+		for _, item := range reply.List {
+			if item.Value == address {
+				found = true
+				sentence = reply
+				fmt.Println(fmt.Sprintf("[DEBUG] Found dhcp lease we were looking for: %v", sentence))
+			}
+		}
+	}
+
+	if !found {
+		return nil, nil
+	}
+
+	// TODO: Add error checking
+
+	macaddress := ""
+	id := ""
+	for _, pair := range sentence.List {
+		if pair.Key == ".id" {
+			id = pair.Value
+		}
+		if pair.Key == "mac-address" {
+			macaddress = pair.Value
+		}
+	}
+
+	return &DhcpLease{
+		Id:      id,
+		MacAddress: macaddress,
+		Address:    address,
+	}, nil
+}
+
+func (client Mikrotik) UpdateDhcpLease(id, address, macaddress string) error {
+	c, err := client.getMikrotikClient()
+
+	if err != nil {
+		return err
+	}
+	cmd := strings.Split(fmt.Sprintf("/ip/dhcp-server/lease/set =numbers=%s =address=%s =mac-address=%s", id, address, macaddress), " ")
+	fmt.Println(fmt.Sprintf("[INFO] Running the mikrotik command: `%s`", cmd))
```

Comment:
In #9 I fixed the provider logging. This should be `log.Printf("[INFO] Running the mikrotik command: %s", cmd)`

Reply:
Updated